File: app/src/main/java/org/breezyweather/background/polling/services/permanent/update/ForegroundNormalUpdateService.java
package org.breezyweather.background.polling.services.permanent.update;

import android.content.Context;

import java.util.List;

import dagger.hilt.android.AndroidEntryPoint;
import org.breezyweather.BreezyWeather;
import org.breezyweather.common.basic.models.Location;
import org.breezyweather.remoteviews.NotificationHelper;
import org.breezyweather.remoteviews.WidgetHelper;
import org.breezyweather.background.polling.services.basic.ForegroundUpdateService;
import org.breezyweather.background.polling.services.permanent.PermanentServiceHelper;

/**
 * Foreground normal update service.
 * */

@AndroidEntryPoint
public class ForegroundNormalUpdateService extends ForegroundUpdateService {

    @Override
    public void updateView(Context context, Location location) {
        WidgetHelper.updateWidgetIfNecessary(context, location);
    }

    @Override
    public void updateView(Context context, List<Location> locationList) {
        WidgetHelper.updateWidgetIfNecessary(context, locationList);
        NotificationHelper.updateNotificationIfNecessary(context, locationList);
    }

    @Override
    public void handlePollingResult(boolean failed) {
        PermanentServiceHelper.updatePollingService(this, failed);
    }

    @Override
    public int getForegroundNotificationId() {
        return BreezyWeather.NOTIFICATION_ID_UPDATING_NORMALLY;
    }
}

File: app/src/main/java/org/breezyweather/background/polling/services/permanent/update/ForegroundNormalUpdateService.kt
package org.breezyweather.background.polling.services.permanent.update

import android.content.Context
import dagger.hilt.android.AndroidEntryPoint
import org.breezyweather.background.polling.services.basic.ForegroundUpdateService
import org.breezyweather.background.polling.services.permanent.PermanentServiceHelper
import org.breezyweather.common.basic.models.Location
import org.breezyweather.remoteviews.Notifications
import org.breezyweather.remoteviews.Widgets

/**
 * Foreground normal update service.
 */
@AndroidEntryPoint
class ForegroundNormalUpdateService : ForegroundUpdateService() {
    override fun updateView(context: Context, location: Location) {
        Widgets.updateWidgetIfNecessary(context, location)
    }

    override fun updateView(context: Context, locationList: List<Location>) {
        Widgets.updateWidgetIfNecessary(context, locationList)
        Notifications.updateNotificationIfNecessary(context, locationList)
    }

    override fun handlePollingResult(updateSucceed: Boolean) {
        PermanentServiceHelper.updatePollingService(this, updateSucceed)
    }

    override val foregroundNotificationId = Notifications.ID_UPDATING_WIDGET
}
